lib/promotion.rb

```diff
@@ -33,7 +33,11 @@ def review_tl0
   end
 
   def review_tl1
-    Promotion.tl2_met?(@user) && change_trust_level!(TrustLevel[2])
+    if Promotion.tl2_met?(@user) && change_trust_level!(TrustLevel[2])
+      SiteSetting.send_tl2_promotion_message && Jobs.enqueue(:send_system_message, user_id: @user.id, message_type: "tl2_promotion_message")
```

Comment:
Let's move the logic here into the `user` model just like how we do for the other welcome messages.